Android/app/src/main/java/io/github/project_travel_mate/login/LoginActivity.java

```diff
@@ -132,11 +138,28 @@ public void onClick(View view) {
                 emailString = email_signup.getText().toString();
                 passString = pass_signup.getText().toString();
                 String confirmPassString = confirm_pass_signup.getText().toString();
-                String nameString = name.getText().toString();
-                if (passString.equals(confirmPassString)) {
-                    mLoginPresenter.ok_signUp(nameString, emailString, passString, mHandler);
-                } else {
-                    Toast.makeText(this, R.string.passwords_check,
+                String firstname = firstName.getText().toString();
+                String lastname = lastName.getText().toString();
+                if (!emailString.equals("") && isValidEmail(emailString)){
+                    if (passString.length()>=8) {
```

Comment:
The if-else loop looks quite complicated. Move it to a different function & simplify it.